src/uct/tcp/tcp_iface.c

```diff
@@ -237,12 +239,19 @@ static ucs_status_t uct_tcp_iface_flush(uct_iface_h tl_iface, unsigned flags,
                                         uct_completion_t *comp)
 {
     uct_tcp_iface_t *iface = ucs_derived_of(tl_iface, uct_tcp_iface_t);
+    uct_tcp_ep_t *ep, *tmp_ep;
 
     if (comp != NULL) {
         return UCS_ERR_UNSUPPORTED;
     }
 
     if (iface->outstanding != 0) {
+        if (ucs_unlikely(!ucs_list_is_empty(&iface->unconn_ep_list))) {
```

Comment:
minor: ```ucs_list_is_empty``` is not needed here because ```ucs_list_for_each_safe``` checks for empty list

Reply:
yes, this is just for optimization reason (check for emptiness with `unlikely`)